frontend_react/src/shared/ui/components/custom/rive-icon.tsx

```diff
@@ -0,0 +1,60 @@
+import { useEffect } from 'react';
+import { useRive, useStateMachineInput } from '@rive-app/react-canvas';
+
+export type RiveIconProps = {
+  /** クラス名 */
+  className?: string;
+  /**
+   * アイコンタイプ
+   * Specifies the type of icon to display. Each type corresponds to a specific animation and state machine.
+   */
+  iconType: 'Home' | 'ToDo' | 'Diary' | 'Account' | 'PLUS';
```

Comment:
[nits]

```suggestion
  iconType: keyof typeof ICON_TYPES
```

（定義する位置の入れ替えが必要かも）

Reply:
ありがとうございます！
提案いただいたコードに変更しました！！